pkg/template/resource.go

```diff
@@ -136,13 +135,13 @@ func applyParamToResourceTemplate(param triggersv1.Param, rt json.RawMessage, ol
 	return bytes.Replace(rt, []byte(paramVariable), []byte(param.Value), -1)
 }
 
-// UID generates a random string like the Kubernetes apiserver generateName metafield postfix.
-var UID = func() string { return rand.String(5) }
+// UUID generates a Universally Unique IDentifier following RFC 4122.
+var UUID = func() string { return uuid.New().String() }
```

Comment:
I have currently changed the naming everywhere from `uid` to `uuid` - I am not certain if this is really required.

`uuid` is more clear as to what it is.
`uid` is a less breaking change (and is the name k8s uses AFAIK).

Reply:
cc @wlynch in case you have suggestions.

Reply:
I agree - I'm leaning towards leaving things as `uid` based on your reasoning + so we don't have to bother with deprecation.

Reply:
This is done now - the change is not only very internal, renaming only the internal functions properly.